battery_checker.cpp

```diff
@@ -0,0 +1,11 @@
+#include <assert.h>
+
+#include "battery_functional_validation.h"
+
+int main()
+{
+  battery::BatteryHealth batteryStatus;
+  assert(batteryStatus.batteryHealthIsOk(25, 70, 0.7) == true);
```

Comment:
These asserts don't cover all conditions in your code. Try testing all possibilities of `checkRange` first. Then you only need to check if the combination of conditions is ok.